WordPress/src/main/java/org/wordpress/android/ui/stockmedia/StockMediaPickerActivity.java

```diff
@@ -395,17 +400,43 @@ public void onStockMediaUploaded(MediaStore.OnStockMediaUploaded event) {
             ToastUtils.showToast(this, R.string.media_upload_error);
             AppLog.e(AppLog.T.MEDIA, "An error occurred while uploading stock media");
         } else {
-            ArrayList<Integer> idList = new ArrayList<>();
-            for (MediaModel media : event.mediaList) {
-                idList.add(media.getId());
+            trackUploadedStockMediaEvent(event.mediaList);
+
+            int count = event.mediaList.size();
+            long[] idArray = new long[count];
+            for (int i = 0; i < count; i++) {
+                idArray[i] = event.mediaList.get(i).getMediaId();
             }
+
             Intent intent = new Intent();
-            intent.putIntegerArrayListExtra(KEY_UPLOADED_MEDIA_IDS, idList);
+            intent.putExtra(KEY_UPLOADED_MEDIA_IDS, idArray);
             setResult(RESULT_OK, intent);
             finish();
         }
     }
 
+    private void trackUploadedStockMediaEvent(@NonNull List<MediaModel> mediaList) {
+        if (mediaList.size() == 0) {
+            AppLog.e(AppLog.T.MEDIA, "Cannot track uploaded stock media event if mediaList is empty");
+            return;
+        }
+
+        boolean isMultiselect = mediaList.size() > 1;
+
+        for (MediaModel media : mediaList) {
+            if (media.getUrl() != null) {
+                Uri mediaUri = Uri.parse(media.getUrl());
+                Map<String, Object> properties =
```

Comment:
Sorry Nick, I suggest you to track media details but noticed that our utility routine doesn't work for remote files. I suggest to remove this call here, and create just an empty properties Map. Maybe adding a TODO to improve the routine to take in consideration remote files as well as local files.

Other than that I think it's ok and ready to be merged.